src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -197,8 +197,15 @@ object EitherDecoder {
 
 sealed trait Setter[+T]
 sealed trait NonNullableSetter[+T]
+sealed trait NonNullableIterableSetter[+T]
 final case class SetValue[+T](set: T) extends Setter[T] with NonNullableSetter[T]
 final case class SetNull[+T]() extends Setter[T]
+// For add/remove on updates for assets_ids, labels (array types)
+final case class AddRemoveArr[+T](add: Seq[T] = Seq.empty, remove: Seq[T] = Seq.empty)
+    extends NonNullableSetter[Seq[T]]
```

Comment:
I think you should extend `NonNullableIterableSetter` here.

In many cases where we have NonNullableSetter, it does not make sense to use `add: ...`, so we should have a different trait which allows `add`, `remove` or `set`

Reply:
to restrict others to not support add/remove, what advantage would it give us?

Reply:
`NonNullableIterableSetter` is a leftover from my back and forths with this but you seem to like that approach better 😅